### ISSUE TITLE
When Kubernetes apiserver down , Traefik 2 starts returning 404

### ISSUE THREAD

**Issue by wu0407** [traefik/traefik#8066]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

To verify traefik ingress controller works correctly  when all kubernetes apiserver is down. 
Traefik work proper a period of time, then return 404.
I see it's fixed in  treafik v1 https://github.com/traefik/traefik/pull/1295

### What did you expect to see?
No matter what circumstances return to 200.

### What did you see instead?
Return 200  a period of time and return to 404.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.4.8
Codename:     livarot
Go version:   go1.16.2
Built:        2021-03-23T15:48:39Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
spec:
  containers:
  - args:
    - --global.checknewversion
    - --global.sendanonymoususage
    - --entryPoints.traefik.address=:9000/tcp
    - --entryPoints.web.address=:8000/tcp
    - --entryPoints.websecure.address=:8443/tcp
    - --api.dashboard=true
    - --ping=true
    - --providers.kubernetescrd
    - --providers.kubernetesingress
    - --api.insecure=true
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
E0414 04:40:47.679821       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: Get "https://10.248.252.1:443/apis/traefik.containo.us/v1alpha1/serverstransports?limit=500&resourceVersion=0": dial tcp 10.248.252.1:443: connect: connection refused
E0414 04:41:18.969402       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: Get "https://10.248.252.1:443/apis/traefik.containo.us/v1alpha1/serverstransports?limit=500&resourceVersion=0": dial tcp 10.248.252.1:443: connect: connection refused
E0414 04:41:58.696359       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: Get "https://10.248.252.1:443/apis/traefik.containo.us/v1alpha1/serverstransports?limit=500&resourceVersion=0": dial tcp 10.248.252.1:443: connect: connection refused
E0414 04:42:43.657032       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.2/tools/cache/reflector.go:167: Failed to watch *v1alpha1.ServersTransport: failed to list *v1alpha1.ServersTransport: Get "https://10.248.252.1:443/apis/traefik.containo.us/v1alpha1/serverstransports?limit=500&resourceVersion=0": dial tcp 10.248.252.1:443: connect: connection refused
```

**Comment by SantoDE** [comment]
Hey @wu0407,

is that for all k8s providers or only for a specific, like Ingress or CRD?

**Comment by wu0407** [comment]
I only test ingress, but crd may same

**Comment by jakubhajek** [comment]
Hey @wu0407 

Thanks for using Traefik and reporting the issue you were facing. 

I was able to reproduce the issue you described but for the Kubernetes Ingress provider. It is not replicable for CRD, which means that I am still getting 200 if Kubernetes API is unreachable.

Steps to reproduce:
1. Run the Kubernetes cluster (1.19.3) and deploy Traefik Proxy (2.4.7). 
2. Deploy a test application and expose it using Ingress and IngressRoute (CRD). 
3. Make the Kubernetes API unreachable e.g. by killing the process. 
4. Wait a few minutes (~10')
5. After that you will 404 for the service that uses Ingress. The service that uses Ingressrouter won't be affected. 
6. Make the API back operable and the Ingress will be again accessible. 

```
E0415 09:19:00.515942       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Endpoints: unknown (get endpoints)
E0415 09:19:00.516175       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.IngressRouteUDP: unknown (get ingressrouteudps.traefik.containo.us)
E0415 09:19:00.516217       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Endpoints: unknown (get endpoints)
E0415 09:19:00.516246       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.TLSOption: unknown (get tlsoptions.traefik.containo.us)
E0415 09:19:00.516272       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.TraefikService: unknown (get traefikservices.traefik.containo.us)
E0415 09:19:00.516295       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.ServersTransport: unknown (get serverstransports.traefik.containo.us)
E0415 09:19:00.516326       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Secret: unknown (get secrets)
E0415 09:19:00.516352       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Service: unknown (get services)
E0415 09:19:00.516378       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1beta1.IngressClass: unknown (get ingressclasses.networking.k8s.io)
E0415 09:19:00.516402       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.Middleware: unknown (get middlewares.traefik.containo.us)
E0415 09:19:00.516423       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Secret: unknown (get secrets)
E0415 09:19:00.516446       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1beta1.Ingress: unknown (get ingresses.extensions)
E0415 09:19:00.516468       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.Service: unknown (get services)
E0415 09:19:00.516490       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.IngressRoute: unknown (get ingressroutes.traefik.containo.us)
E0415 09:19:00.516510       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.TLSStore: unknown (get tlsstores.traefik.containo.us)
E0415 09:19:00.516532       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1alpha1.IngressRouteTCP: unknown (get ingressroutetcps.traefik.containo.us)
```
```json
{"level":"error","msg":"Failed to get server version: could not retrieve server version: Get \"https://10.233.0.1:443/version?timeout=32s\": dial tcp 10.233.0.1:443: connect: connection refused","providerName":"kubernetes","time":"2021-04-15T09:16:18Z"}
{"level":"error","msg":"Failed to get server version: could not retrieve server version: Get \"https://10.233.0.1:443/version?timeout=32s\": dial tcp 10.233.0.1:443: connect: connection refused","providerName":"kubernetes","time":"2021-04-15T09:16:18Z"}
```

**Comment by traefiker** [comment]
Closed by #8185.
